cinema/tests/test_movie_view_set.py

```diff
@@ -0,0 +1,201 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from rest_framework import status
+from rest_framework.reverse import reverse
+from rest_framework.test import APIClient
+
+from cinema.models import Movie, Genre, Actor
+from cinema.serializers import MovieSerializer, MovieListSerializer, MovieDetailSerializer
+
+MOVIE_URL = reverse("cinema:movie-list")
+
+def detail_url(movie_id):
+    return reverse("cinema:movie-detail", args=[movie_id])
+
+def sample_movie(**params) -> Movie:
+    defaults = {
+        "title": "The Great Brick",
+        "description": "The Great Brick fall on superman head",
+        "duration": 121
+    }
+    defaults.update(params)
+    return Movie.objects.create(**defaults)
+
+
+class TestUnauthenticatedUser(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_auth_required(self):
+        res = self.client.get(MOVIE_URL)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class TestAuthenticatedUser(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="user1@cinema.com",
+            password="2wsxcvfr4",
+        )
+        self.client.force_authenticate(self.user)
+
+        self.movie = sample_movie()
+        self.comedy_movie = sample_movie(title="The Great Brick 2")
+        self.drama_movie = sample_movie(title="The Great Brick 3")
+
+        self.genre_1 = Genre.objects.create(name="Comedy")
+        self.genre_2 = Genre.objects.create(name="Drama")
+
+        self.actor_1 = Actor.objects.create(
+            first_name="Adam",
+            last_name="Smith",
+        )
+        self.actor_2 = Actor.objects.create(
+            first_name="Eva",
+            last_name="Smith",
+        )
+
+        self.comedy_movie.genres.add(self.genre_1)
+        self.drama_movie.genres.add(self.genre_2)
+
+        self.comedy_movie.actors.add(self.actor_1)
+        self.drama_movie.actors.add(self.actor_2)
+
+        self.payload = {
+            "title": "The Great Brick 4",
+            "description": "The Great Brick fall on superman head",
+            "duration": 121,
+        }
+
+    def test_movie_list(self):
+        self.client.force_authenticate(self.user)
+        res = self.client.get(MOVIE_URL)
+        movies = Movie.objects.all()
+        serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+
+        self.assertEqual(res.data, serializer.data)
```

Comment:
This is a code quality issue. You're using the `assertEqual` method to compare the response data with the serialized data. This is a good practice as it ensures that the API returns the expected data.